packages/flutter_tools/lib/src/commands/screenshot.dart

```diff
@@ -183,6 +191,38 @@ class ScreenshotCommand extends FlutterCommand {
     return true;
   }
 
+  /// Additional error checks below for valid file name
+  /// and directory because ios/xcode screenshot binary will
+  /// return a 0 exit code even if outputFile is not valid
+  static void validateOutputFile(File outputFile, FileSystem fs) {
+
+    const Platform platform = LocalPlatform();
+
+    // Conditional to check for trailing path separator
+    if (outputFile.path.endsWith(platform.pathSeparator)) {
+      throwToolExit(
+        'The provided path cannot end with a path separator\n'
+        'Path provided: "${outputFile.path}"'
+      );
+    }
+
+    // Conditional for validating directory is valid
+    if (!fs.directory(outputFile.dirname).existsSync()) {
+      throwToolExit(
+        'The provided path to file needs to have a directory that exists\n'
+        'Directory: "${outputFile.dirname}" does not exist'
+      );
+    }
```

Comment:
Use `fs.file(outputFile).parent.existsSync()` - I'm not sure if dirname includes the contents of absolute paths.